source/globals.d.ts

```diff
@@ -1,8 +1,7 @@
 /* eslint-disable @typescript-eslint/consistent-indexed-object-style */
 
 type AnyObject = Record<string, any>;
-// TODO: Fix this type (there's no such `delegate` global type) and enable the @typescript-eslint/no-redundant-type-constituents rule
-type Deinit = delegate.Subscription | Observer | MutationObserver | ResizeObserver | IntersectionObserver | VoidFunction;
+type Deinit = MutationObserver | ResizeObserver | IntersectionObserver | {destroy: VoidFunction} | {abort: VoidFunction} | VoidFunction;
```

Comment:
If you turn this into an [arrayable](https://github.com/sindresorhus/type-fest/issues/270) you never have to write `Deinit[]`

Reply:
> If you turn this into an arrayable you never have to write `Deinit[]`

+1

> Perhaps if you include a void in Deinit itself we can just always return Deinit for every init function

I don't know, I kinda like the fact that `void` makes it obvious the init returns nothing. Plus it's easier to grep for only features that have a specific deinit mechanism.